File: TypeScript/practice01.ts

let userName: string = 'Alice'; // 型指定
const age: number = 25; // 定数
let isStudent = true; // 型推論

let message: string = 'Hello, TypeScript!'; // 文字列
let count: number = 42; // 数値
let isDone: boolean = true; // 真偽値
let something = "TypeScript"; // 型推論：文字列

// 配列
let numbers: number[] = [1, 2, 3];
let strings: string[] = ["apple", "banana", "cherry"];

// タプル
let tuple: [string, number] = ["Alice", 25];

// 関数
function greet(name: string): string {
  return `Hello, ${name}!`;
}

function logMessage(message: string): void {
  console.log(message);
}

logMessage(greet(userName));

// オブジェクト
let person: { name: string, age: number } = {
    name: 'Alice',
    age: 25
};

console.log(person);

// クラス
class Animal {
    name: string;

    constructor(name: string) {
        this.name = name;
    }

    speak(): void {
        console.log(`${this.name} makes a sound.`);
    }
}

let dog = new Animal('Dog');
dog.speak();

// 型のユニオン
let value: string | number;
value = "Hello";
value = 42;

// 型エイリアス
type ID = string | number;
let userID: ID = 123;
userID = "ABC";

// 型アサーション
let someValue: any = "Hello, TypeScript!";
let stringLength: number = (someValue as string).length;

function add(a: number, b: number): number {
    return a + b;
}
console.log(add(1, 2)); // 3

function toString(value: number | string): string {
    return value.toString();
}
console.log(toString(42)); // "42"

type member = { name: string, age: number };
let mem: member = { name: 'Bob', age: 30 };
console.log(mem);

File: 2025/1/practice01.c
#include <stdio.h>

int main()
{
  // 変数の宣言と初期化
  int age = 25;
  float pi = 3.14;

  // 制御構造

  // if文
  int num = 10;

  if (num > 5)
  {
    printf("numは5より大きい\n");
  }
  else
  {
    printf("numは5以下\n");
  }

  // for文
  for (int i = 0; i < 5; i++)
  {
    printf("iの値: %d\n", i);
  }

  // while文
  int count = 0;
  while (count < 3)
  {
    printf("count: %d\n", count);
    count++;
  }

  // switch文
  char grade = 'B';

  switch (grade)
  {
  case 'A':
    printf("素晴らしい！\n");
    break;
  case 'B':
    printf("良い成績！\n");
    break;
  default:
    printf("もっと頑張ろう！\n");
  }

  // 配列
  int numbers[5] = {1, 2, 3, 4, 5};
  printf("numbers[1]: %d\n", numbers[1]);

  // 文字列
  char name[] = "Alice";
  printf("name: %s\n", name);
}